activation/activation_test.go

```diff
@@ -1487,3 +1485,99 @@ func TestFindFullyValidHighTickAtx(t *testing.T) {
 		require.Equal(t, atxLower.ID(), found)
 	})
 }
+
+// Test_Builder_RegenerateInitialPost tests the coverage for the edge case
+// where a node operator may change SUs after creating the initial PoST but before
+// submitting the first ATX, which should result in the initial PoST to be deleted
+// and for the new PoST to be generated instead (this also loses the eligibility
+// for the current epoch). This behavior is mocked by mocking the response of the
+// proof validator.
+func Test_Builder_RegenerateInitialPost(t *testing.T) {
+	var (
+		tab           = newTestBuilder(t, 1)
+		sig           = maps.Values(tab.signers)[0]
+		genesis       = time.Now()
+		ctx, cancel   = context.WithCancel(context.Background())
+		commitmentATX = types.RandomATXID()
+		nonce         = types.VRFPostIndex(rand.Uint64())
+		numUnits      = uint32(12)
+		initialPost   = &types.Post{
+			Nonce:   rand.Uint32(),
+			Indices: types.RandomBytes(10),
+			Pow:     rand.Uint64(),
+		}
+	)
+
+	postV2Calls := 0
+	tab.mValidator.EXPECT().
+		PostV2(gomock.Any(), sig.NodeID(), commitmentATX, initialPost, shared.ZeroChallenge, numUnits).DoAndReturn(func(_ context.Context, _ types.NodeID, _ types.ATXID, _ *types.Post, _ []byte, _ uint32, _ ...validatorOption) error {
+		switch postV2Calls {
+		case 0, 1, 3:
+			postV2Calls++
+			return nil
+		case 2:
+			postV2Calls++
+			return nil
+		}
+		panic("unexpected test branch")
+	}).Times(4)
```

Comment:
Whats the difference between the 3rd call and the others? Would this also work?

```suggestion
	tab.mValidator.EXPECT().
		PostV2(gomock.Any(), sig.NodeID(), commitmentATX, initialPost, shared.ZeroChallenge, numUnits).Times(4)
```